crates/eth-rpc/src/servers/web3_rpc.rs

```diff
@@ -0,0 +1,27 @@
+use jsonrpsee::core::{async_trait, RpcResult as Result};
+use kakarot_rpc_core::client::constants::KAKAROT_CLIENT_VERSION;
+use reth_primitives::{keccak256, Bytes, H256};
+
+use crate::api::web3_api::Web3ApiServer;
+
+/// The RPC module for the implementing Web3 Api { i.e rpc endpoints prefixed with web3_ }
+#[derive(Default)]
+pub struct Web3Rpc {}
+
+impl Web3Rpc {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+#[async_trait]
+impl Web3ApiServer for Web3Rpc {
+    fn client_version(&self) -> Result<String> {
+        Ok(String::from(KAKAROT_CLIENT_VERSION))
+    }
+
+    fn sha3(&self, input: Bytes) -> Result<H256> {
```

Comment:
For now we just return Keccak256, but we should in future investigate how tools use this endpoint, and according to that what hash should we actually returns { given we use a mix of hashing algorithms afaik }

Reply:
Also, do we think we can make a low priority issues for the above , if yes then I can create that.

Reply:
I think that there has been a long confusion in the two terms and eth eventually migrated to keccak256, maybe this route is old? https://ethereum.stackexchange.com/questions/550/which-cryptographic-hash-function-does-ethereum-use

Reply:
Oh, so what I am talking about here is that in Ethereum, hashing is Keccak256, so for L1 execution client, this endpoint should return Keccak256.

In zk-evms, usually we do use zk-friendly hashing algorithms , from the daily we had I got to know that we use a mix of hashing algorithm.

So, I think we should investigate that the tools that use this endpoint, what do they usually use it for, and then see if there is some other hashing algorithm that we need to use here. { although this should not be a priority, as we aren't sure how much is this endpoint used. }

Lemme know if you need more clarity.